junebug/channel.py

```diff
@@ -0,0 +1,147 @@
+import collections
+from copy import deepcopy
+import json
+import uuid
+from twisted.internet.defer import inlineCallbacks, returnValue
+from twisted.web import http
+from vumi.service import WorkerCreator
+from vumi.servicemaker import VumiOptions
+from vumi.persist.redis_manager import RedisManager
+
+from junebug.error import JunebugError
+
+
+class ChannelNotFound(JunebugError):
+    '''Raised when a channel's data cannot be found.'''
+    name = 'ChannelNotFound'
+    description = 'channel not found'
+    code = http.NOT_FOUND
+
+
+class InvalidChannelType(JunebugError):
+    '''Raised when an invalid channel type is specified'''
+    name = 'InvalidChannelType',
+    description = 'invalid channel type'
+    code = http.BAD_REQUEST
+
+
+transports = {
+    'telnet': 'vumi.transports.telnet.TelnetServerTransport',
+    'xmpp': 'vumi.transports.xmpp.XMPPTransport',
+}
+
+
+class Channel(object):
+    def __init__(
+            self, redis_config, amqp_config, properties, id=None):
+        '''Creates a new channel. ``redis_config`` is the redis config, from
+        which a sub manager is created using the channel id. If the channel id
+        is not supplied, a UUID one is generated. Call ``save`` to save the
+        channel data. It can be started using the ``start`` function.'''
+        self._properties, self.id = properties, id
+        if self.id is None:
+            self.id = str(uuid.uuid4())
+        self._redis_base = RedisManager.from_config(redis_config)
```

Comment:
Ask discussed at the office, this needs to be a TxRedisManager (just noting it here).